Source/Core/Common/UnixUtil.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <pthread.h>
+#include <sys/eventfd.h>
+
+#include "Common/CommonFuncs.h"
+#include "Common/Logging/Log.h"
+
+namespace UnixUtil
+{
+inline int CreateEventFD(unsigned int count, int flags)
+{
+  const int result = eventfd(count, flags);
+  if (result == -1)
+  {
+    ERROR_LOG_FMT(COMMON, "eventfd failed: {}", Common::LastStrerrorString());
+    std::abort();
+  }
+  return result;
+}
+
+// Repeatedly call a function that can erroneously produce EINTR.
+auto RetryOnEINTR(auto func, auto... args)
```

Comment:
Are functions that use auto parameters like this automatically `inline` because they're technically templates?

Reply:
Yeah: https://en.cppreference.com/w/cpp/language/function_template.html#Abbreviated_function_template

Reply:
I suppose they must be based on [this description](https://en.cppreference.com/w/cpp/language/function_template.html#Abbreviated_function_template).